pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -194,6 +221,67 @@ func (r *ReconcileCheInstallation) ensureCheSubscription(logger logr.Logger, che
 	return false, nil
 }
 
+func (r *ReconcileCheInstallation) ensureWatchCheCluster() error {
+	if r.watchCheCluster != nil {
+		cheClusterCRD := &apiextnv1beta1.CustomResourceDefinition{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "checlusters.org.eclipse.che"}, cheClusterCRD); err != nil {
+			return err
+		}
+		if err := r.watchCheCluster(); err != nil {
+			return err
+		}
+		r.watchCheCluster = nil // make sure watchCheCluster() should NOT be called afterwards
```

Comment:
Well... let's make `ensureWatchCheCluster()` thread safe. Just in case multiple go routines will start reconcile.
Let's lock when we are entering the function and unlock in defer:
```
func (r *ReconcileCheInstallation) ensureWatchCheCluster() error {
	r.mux.Lock()
	if r.watchCheCluster != nil {
	    ...
	}
	defer r.mux.Unlock()
}
```

Reply:
Thanks @alexeykazakov  .. Done https://github.com/codeready-toolchain/toolchain-operator/pull/17/commits/8cf2914a793fa90ffa448f261fa0e3cf3515064b